.travis.yml

```diff
@@ -0,0 +1,53 @@
+language: scala
+scala:
+  - 2.12.8
+
+os:
+  - osx
+  - linux
+
+addons:
+  homebrew:
+    packages:
+    - gmp
+    - re2
+    - pkgconfig
+    - python3
+    - libtool
+    - protobuf
+  apt:
+    packages:
+    - libre2-dev
+    - python3-pip
+    - makedev
+    - fuse
+    - libfuse-dev
+    - libsqlite3-dev
+    - libgmp-dev
+    - libncurses5-dev
+    - pkg-config
+    - git
+    - g++
+    - gcc
+    - libtool-bin
+    - protobuf-compiler
+
+install:
+  - mkdir -p repos/block-inclusivecache-sifive
+  - git clone $PWD repos/block-inclusivecache-sifive
+  - cd repos
+  - git clone https://github.com/chipsalliance/rocket-chip.git
+  - git clone https://github.com/ucb-bar/berkeley-hardfloat.git
+  - git clone https://github.com/sifive/api-chisel3-sifive.git
+  - git clone https://github.com/freechipsproject/chisel3.git
+  - git clone https://github.com/sifive/api-firrtl-sifive.git
+  - git clone https://github.com/sifive/api-scala-sifive.git
+  - git clone https://github.com/sifive/wit.git
+  - git clone https://github.com/freechipsproject/firrtl.git
+  - git clone https://github.com/sifive/rocket-chip-wake.git
```

Comment:
Can we use wit itself to check out all these other things? Also, I think most of these should be replaced with `api-generator-sifive` and `soc-testsocket-sifive`. The other benefit of using wit is that we would then be able to pin to specific versions of these repos; without that, this code will break as soon as someone merges in a backwards incompatible change to the master branch of any of these repos.

Reply:
The issue is that it wit tries to use ssh by default which Travis can't do. We really need a fix for this in wit.

Per pinned versions, wit will still resolve them just fine actually, even if the master pointer moves, `--repo-path` just has wit use an alternative remote to clone from, but it will still check out the correct commits

Reply:
OK, well, can we at least add a comment describing why this is like this? Also, I think this is a pretty high priority thing for us to fix in `wit`.

Reply:
Heads up that git is able to rewrite URLS.

For example, to globally rewrite Github SSH to Github HTTPS:
```
git config --global url."https://github.com/".insteadOf 'git@github.com:'
```

Reply:
cool I'll try that. thanks!

Reply:
Also, this is what the change to the ~/.gitconfig looks like:

```ini
[url "https://github.com/"]
	insteadOf = git@github.com:
```